src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -180,13 +180,22 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
         };
 
         await kubernetesService.CreateAsync(dashboardExecutable, cancellationToken).ConfigureAwait(false);
+        await WaitForHttpSuccessOrThrow(dashboardUrl, GetDashboardWaitTimeInSeconds(), cancellationToken).ConfigureAwait(false);
+    }
 
-        await WaitForHttpSuccessOrThrow(dashboardUrl, TimeSpan.FromSeconds(DashboardWaitTimeInSeconds), cancellationToken).ConfigureAwait(false);
+    private static TimeSpan GetDashboardWaitTimeInSeconds()
```

Comment:
This is returning a `TimeSpan`.

```suggestion
    private static TimeSpan GetDashboardWaitTime()
```

Consider also renaming it so it's clearer what it's waiting for. Maybe `GetDashboardAvailabilityTimeoutDuration` or something like that.